websocket_server/ServerManager.js

```diff
@@ -31,6 +34,76 @@ export default class ServerManager {
 		this.appManager = new AppManager(this.storageManager)
 		this.server = this.createConfiguredServer(this.appManager.getApp())
 		this.socketManager = new SocketManager(this.server, this.roomDataManager, this.storageManager)
+		this.recordingServices = new Map()
+
+		// Setup CORS and Cookie Parser
+		const corsOptions = {
+			origin: process.env.NEXTCLOUD_URL || 'http://nextcloud.local',
+			methods: ['GET', 'POST', 'PUT', 'DELETE', 'OPTIONS'],
+			credentials: true,
+			optionsSuccessStatus: 200,
+		}
+		this.appManager.getApp().use(cors(corsOptions))
+		this.appManager.getApp().use(cookieParser())
+	}
+
+	setupRecordingRoutes() {
+		// Start recording
+		this.appManager.getApp().post('/:roomId/record', async (req, res) => {
+			try {
+				const { roomId } = req.params
+				if (!roomId) {
+					return res.status(400).json({ error: 'Invalid request body: roomId is required and must be a string' })
+				}
+
+				if (this.recordingServices.has(roomId)) {
+					return res.status(400).json({ error: 'Already recording' })
+				}
+
+				if (!req.body || typeof req.body.boardUrl !== 'string') {
+					return res.status(400).json({ error: 'Invalid request body: boardUrl is required and must be a string' })
+				}
+
+				const { boardUrl } = req.body
+
+				const recorder = new RecordingService()
+				const initialized = await recorder.init(boardUrl)
+				if (!initialized) {
+					throw new Error('Init failed')
+				}
+
+				const nc_username = req.cookies.nc_username
```

Comment:
We should not rely on the session cookies here I think. I know this might be moved to websocket communication anyways, but if we keep the separate endpoints we should use the jwt token to authenticate I and get the user id i think.